File: common/src/main/java/com/asiainfo/ocdp/common/router/Router.java
package com.asiainfo.ocdp.common.router;

import java.util.HashMap;
import java.util.Map;

/**
 * <pre>
 *
 * Created by zhenqin.
 * User: zhenqin
 * Date: 16/10/9
 * Time: 11:50
 * Verdor: NowledgeData
 * To change this template use File | Settings | File Templates.
 *
 * </pre>
 *
 * @author zhenqin
 */
public abstract class Router {

    protected final String cacheManagers;



    protected final Map<String, String> PROXY_HOST_MAP = new HashMap<String, String>();


    public Router(String cacheManagers) {
        this.cacheManagers = cacheManagers;
        String[] managers = this.cacheManagers.split(",");
        for (String manager : managers) {
            String[] split = manager.split(":");
            this.PROXY_HOST_MAP.put(split[0], manager);
        }
    }

    public abstract String getProxyHost(String host);
}

File: core/src/main/scala/com/asiainfo/ocdp/stream/common/Router.scala
package com.asiainfo.ocdp.stream.common

import java.util

import com.asiainfo.ocdp.stream.config.MainFrameConf
import redis.clients.jedis.{JedisPool, JedisPoolConfig}


/**
  *
  *
  * <pre>
  * User: liuyu
  * Date: 2016/10/8
  * Time: 16:45
  * </pre>
  *
  * @author liuyu
  */
abstract class Router extends Logging with Serializable{

	var cacheManager: String =""   //连接代理
	val hostMap = new scala.collection.mutable.HashMap[String,util.LinkedList[String]]

	val JedisConfig = new JedisPoolConfig()
	JedisConfig.setMaxTotal(MainFrameConf.systemProps.getInt("jedisPoolMaxTotal"))
	JedisConfig.setMaxIdle(MainFrameConf.systemProps.getInt("jedisPoolMaxIdle"))
	JedisConfig.setMinIdle(MainFrameConf.systemProps.getInt("jedisPoolMinIdle"))
	JedisConfig.setMinEvictableIdleTimeMillis(MainFrameConf.systemProps.getInt("jedisPoolMEM"))
	JedisConfig.setTestOnBorrow(true)

	/*val JedisConfig = new JedisPoolConfig()
	JedisConfig.setMaxTotal(3)
	JedisConfig.setMaxIdle(10)
	JedisConfig.setMinIdle(1)
	JedisConfig.setMinEvictableIdleTimeMillis(1)
	JedisConfig.setTestOnBorrow(true)*/

	def proxyHost(host: String): JedisPool
}
